### PR TITLE
Test speedup

### PR DESCRIPTION
**Pull request by RudolfWeeber**
random pairs and wang landau reaction ensemble

### CODE REVIEW

File: testsuite/random_pairs.py [hunk 1]
```diff
@@ -27,20 +27,32 @@ def setUp(self):
         np.random.seed(2)
 
         s.part.add(pos=s.box_l * np.random.random((n_part, 3)))
+        self.all_pairs=[]
+        parts = self.system.part
```
Comment (KaiSzuttor) [inline testsuite/random_pairs.py]: this variable is not used, please remove

File: testsuite/random_pairs.py [hunk 2]
```diff
@@ -27,20 +27,32 @@ def setUp(self):
...
-        parts = self.system.part
-
-        pairs = []
-        for i in range(len(parts)):
-            for j in range(i + 1, len(parts)):
-                if self.system.distance(parts[i], parts[j]) < dist:
-                    pairs.append((i, j))
-
-        self.assertTrue(len(pairs))
+        # Go through list of all possible pairs for full periodicy
```
Comment (KaiSzuttor) [inline testsuite/random_pairs.py:46]: Could you please turn that into a docstring for the method?

**Review by KaiSzuttor — APPROVED** [review]
LGTM besides comment

**Comment by KaiSzuttor** [comment]
the macs seem to have lower precision? is this compiler related?

**Comment by fweik** [comment]
@KaiSzuttor We've observed this before. One reason could be that the standard library implementation differs (clang vs gcc), and there is some implementation freedom there for the rng distributions (not the rng itself, that has to be identical). As there are other more technical differences (e.g. the handling of memory pages is slightly different on macOS than on Linux), it could also be some subtle bug. Unfortunately this is very hard to investigate, especially without a mac.

**Comment by KaiSzuttor** [comment]
I see, but we also have a clang build on linux...

**Comment by mkuron** [comment]
Clang on Linux still uses libstdc++, so things are expected to match. On Mac, it uses libc++. You could actually try libc++ on Linux, but you would need to recompile Boost.

**Comment by KaiSzuttor** [comment]
what about a gcc build on mac?

**Comment by RudolfWeeber** [comment]
So, I'm not sure the accuracy on Mac is (on average) worse. When it is better, we probably don't notice.

**Pull request merged**